main.cpp

```diff
@@ -0,0 +1,31 @@
+#include "constructors-call.h"
+#include "diamond.h"
+#include "aggregation.h"
+#include "composition.h"
+
+using namespace std;
+
+int main() {
+    // Constructor Call (with parameter)
+    MusicalInstrument my_instrument("instrument");
+    Piano my_piano("piano");
+    Guitar my_guitar("guitar");
+
+    // Diamond problem
+    Baby my_baby;
+    my_baby.Boy::display();
+    my_baby.Girl::display();
+
+    // Aggregation
+    CookTop new_cook_top("KitchenAid", "KCES550HSS");
+    Kitchen new_kitchen(2023, "first floor", &new_cook_top);
```

Comment:
Quiz: What if I pass new_cook_top object without ampersand symbol (line 21). Explore the difference, will discuss in class